### PR TITLE
Add failing test fixture for ParamTypeDeclarationRector

### PR DESCRIPTION
**Pull request by Aerendir**
# Failing Test for ParamTypeDeclarationRector

Based on https://getrector.org/demo/56b352cf-6ba5-4fd5-8186-dd071d1e4010

Refers to https://github.com/rectorphp/rector/issues/5270

### CODE REVIEW

File: rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    /**
+     * @param string $countryCode
+     */
+    public function __construct($countryCode)
+    {
+    }
+}
+?>
+-----
```
Comment (samsonasik) [inline rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc:30]: `-----` can be removed:

```suggestion
```

File: rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+interface VatRateInterface extends SimpleValueObjectInterface
+{
+    /**
+     * @param string $countryCode
+     */
+    public function __construct($countryCode);
+}
+
+final class VatRate implements VatRateInterface
```
Comment (samsonasik) [inline rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc:20]: class name can be changed with `Skip` prefix, eg:

```suggestion
final class SkipDifferentWithInterface implements VatRateInterface
```

and fixture file name can be changed to `skip_different_with_interface.php.inc`

File: rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc (hunk 1 above)
Comment (Aerendir) [reply rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc]: Done!

File: rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc (hunk 2 above)
Comment (Aerendir) [reply rules/type-declaration/tests/Rector/FunctionLike/ParamTypeDeclarationRector/Fixture/vat_rate.php.inc]: Done!

**Comment by samsonasik** [comment]
@Aerendir I cherry-picked your commits at PR https://github.com/rectorphp/rector/pull/5292

**Pull request closed without merging**